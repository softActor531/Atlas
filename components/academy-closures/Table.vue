<template>
  <div class="card">
    <div class="card-header bg-white">
      <h4 class="m-0">
        <i class="mdi mdi-format-list-bulleted mr-1 text-primary" />
        Academy Closures
      </h4>
    </div>
    <div class="card-body">
      <div
        id="DataTables_Table_0_wrapper"
        class="dataTables_wrapper dt-bootstrap4 no-footer"
      >
        <div class="row">
          <div class="col-sm-12 col-md-6">
            <div id="DataTables_Table_0_length" class="dataTables_length">
              <label
                >Show
                <select
                  name="DataTables_Table_0_length"
                  aria-controls="DataTables_Table_0"
                  class="custom-select custom-select-sm form-control form-control-sm"
                  ><option value="25">25</option
                  ><option value="50">50</option
                  ><option value="100">100</option></select
                >
                entries</label
              >
            </div>
          </div>
          <div class="col-sm-12 col-md-6" />
        </div>
        <div class="row">
          <div class="col-sm-12">
            <div class="dataTables_scroll overflow-auto">
              <div
                class="dataTables_scrollHead"
                style=" position: relative; border: 0px; width: 100%;"
              >
                <div
                  class="dataTables_scrollHeadInner"
                  style="box-sizing: content-box; width: 946.4px; padding-right: 17px;"
                >
                  <table
                    class="table table-sm contacts-table clickable zebra closures dataTable no-footer"
                    style="width: 946.4px; margin-left: 0px;"
                    role="grid"
                  >
                    <thead>
                      <tr role="row">
                        <th
                          class="header sorting_asc"
                          tabindex="0"
                          aria-controls="DataTables_Table_0"
                          rowspan="1"
                          colspan="1"
                          style="width: 56px;"
                          aria-sort="ascending"
                          aria-label="District: activate to sort column descending"
                        >
                          District
                        </th>
                        <th
                          class="header sorting"
                          tabindex="0"
                          aria-controls="DataTables_Table_0"
                          rowspan="1"
                          colspan="1"
                          style="width: 60px;"
                          aria-label="Site: activate to sort column ascending"
                        >
                          Site
                        </th>
                        <th
                          class="header sorting"
                          tabindex="0"
                          aria-controls="DataTables_Table_0"
                          rowspan="1"
                          colspan="1"
                          style="width: 46.4px;"
                          aria-label="Name: activate to sort column ascending"
                        >
                          Name
                        </th>
                        <th
                          class="header sorting"
                          tabindex="0"
                          aria-controls="DataTables_Table_0"
                          rowspan="1"
                          colspan="1"
                          style="width: 59.2px;"
                          aria-label="Type: activate to sort column ascending"
                        >
                          Type
                        </th>
                        <th
                          class="header sorting"
                          tabindex="0"
                          aria-controls="DataTables_Table_0"
                          rowspan="1"
                          colspan="1"
                          style="width: 32px;"
                          aria-label="Start Date: activate to sort column ascending"
                        >
                          Start Date
                        </th>
                        <th
                          class="header sorting"
                          tabindex="0"
                          aria-controls="DataTables_Table_0"
                          rowspan="1"
                          colspan="1"
                          style="width: 30.4px;"
                          aria-label="End Date: activate to sort column ascending"
                        >
                          End Date
                        </th>
                        <th
                          class="header sorting"
                          tabindex="0"
                          aria-controls="DataTables_Table_0"
                          rowspan="1"
                          colspan="1"
                          style="width: 32.8px;"
                          aria-label="Start Time: activate to sort column ascending"
                        >
                          Start Time
                        </th>
                        <th
                          class="header sorting"
                          tabindex="0"
                          aria-controls="DataTables_Table_0"
                          rowspan="1"
                          colspan="1"
                          style="width: 32.8px;"
                          aria-label="End Time: activate to sort column ascending"
                        >
                          End Time
                        </th>
                        <th
                          class="header sorting"
                          tabindex="0"
                          aria-controls="DataTables_Table_0"
                          rowspan="1"
                          colspan="1"
                          style="width: 34.4px;"
                          aria-label="All Day?: activate to sort column ascending"
                        >
                          All Day?
                        </th>
                        <th
                          class="header sorting"
                          tabindex="0"
                          aria-controls="DataTables_Table_0"
                          rowspan="1"
                          colspan="1"
                          style="width: 61.6px;"
                          aria-label="Available On-Site?: activate to sort column ascending"
                        >
                          Available On-Site?
                        </th>
                        <th
                          class="header sorting"
                          tabindex="0"
                          aria-controls="DataTables_Table_0"
                          rowspan="1"
                          colspan="1"
                          style="width: 64px;"
                          aria-label="Available Virtually?: activate to sort column ascending"
                        >
                          Available Virtually?
                        </th>
                        <th
                          class="header sorting"
                          tabindex="0"
                          aria-controls="DataTables_Table_0"
                          rowspan="1"
                          colspan="1"
                          style="width: 51.2px;"
                          aria-label="Active?: activate to sort column ascending"
                        >
                          Active?
                        </th>
                        <th
                          class="header sorting_disabled"
                          rowspan="1"
                          colspan="1"
                          style="width: 21.6px;"
                          aria-label=""
                        >
                          Actions
                        </th>
                      </tr>
                    </thead>
                  </table>
                </div>
              </div>
              <div
                class="dataTables_scrollBody"
                style="position: relative; width: 100%;"
              >
                <table
                  id="DataTables_Table_0"
                  class="table table-sm contacts-table clickable zebra closures dataTable no-footer"
                  style="width: 100%;"
                  role="grid"
                  aria-describedby="DataTables_Table_0_info"
                >
                  <tbody>
                    <div
                      v-for="(closure, index) in academyClosures"
                      :key="index"
                    >
                      <tr
                        role="row"
                        class="odd"
                        v-for="site in closure.sites"
                        :key="site.id"
                      >
                        <td class="sorting_1">
                          {{ site.district.name }}
                        </td>
                        <td>
                          {{ site.name }}
                        </td>
                        <td>{{ closure.name }}</td>
                        <td>{{ closure.closure_type }}</td>
                        <td>{{ closure.start_date }}</td>
                        <td>{{ closure.end_date }}</td>
                        <td>{{ closure.start_time }}</td>
                        <td>{{ closure.end_time }}</td>
                        <td class="pl-2 pr-2">
                          {{ closure.all_day ? "true" : "false" }}
                        </td>
                        <td class="pl-2 pr-2">
                          {{ closure.available_on_site ? "true" : "false" }}
                        </td>
                        <td class="pl-3 pr-2">
                          {{ closure.available_virtually ? "true" : "false" }}
                        </td>
                        <td class="pl-4 pr-3">
                          <i
                            :class="
                              `icon active fas fa-circle
                           $  {closure.active ? 'text-success' : ''}
                           `
                            "
                          />
                        </td>
                        <td>
                          <a
                            class="btn btn-primary btn-xs btn-confirmed"
                            @click.prevent="editAppointment(1)"
                            href="#"
                          >
                            <i class="mdi mdi-square-edit-outline" />
                          </a>
                          <a
                            class="btn btn-danger btn-xs btn-confirmed btn-delete-closure"
                            data-closure-id="56"
                            href="#"
                          >
                            <i class="fa fa-trash" />
                          </a>
                        </td>
                      </tr>
                    </div>
                  </tbody>
                </table>
              </div>
            </div>
          </div>
        </div>
        <div class="row">
          <div class="col-sm-12 col-md-5">
            <div
              id="DataTables_Table_0_info"
              class="dataTables_info"
              role="status"
              aria-live="polite"
            >
              Showing 1 to 25 of 28 entries
            </div>
          </div>
          <div class="col-sm-12 col-md-7">
            <div
              id="DataTables_Table_0_paginate"
              class="dataTables_paginate paging_simple_numbers"
            >
              <ul class="pagination pagination-rounded">
                <li
                  id="DataTables_Table_0_previous"
                  class="paginate_button page-item previous disabled"
                >
                  <a
                    href="#"
                    aria-controls="DataTables_Table_0"
                    data-dt-idx="0"
                    tabindex="0"
                    class="page-link"
                    ><i class="mdi mdi-chevron-left"
                  /></a>
                </li>
                <li class="paginate_button page-item active">
                  <a
                    href="#"
                    aria-controls="DataTables_Table_0"
                    data-dt-idx="1"
                    tabindex="0"
                    class="page-link"
                    >1</a
                  >
                </li>
                <li class="paginate_button page-item ">
                  <a
                    href="#"
                    aria-controls="DataTables_Table_0"
                    data-dt-idx="2"
                    tabindex="0"
                    class="page-link"
                    >2</a
                  >
                </li>
                <li
                  id="DataTables_Table_0_next"
                  class="paginate_button page-item next"
                >
                  <a
                    href="#"
                    aria-controls="DataTables_Table_0"
                    data-dt-idx="3"
                    tabindex="0"
                    class="page-link"
                    ><i class="mdi mdi-chevron-right"
                  /></a>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</template>

<script>
export default {
  name: "Table",

  props: {
    academyClosures: {
      type: Object,
      default: {}
    }
  },

  methods: {
    editAppointment(closureId) {
      const path = `/academy-closures/${closureId}/edit`;
      this.$router.push({ path });
    }
  }
};
</script>

<style>
.hide {
  display: none;
}
td {
  min-width: 60px;
}
</style>
