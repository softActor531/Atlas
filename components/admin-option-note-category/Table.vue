<template>
  <div class="card mt-3 p-3">
    <div class="row">
      <div class="col-md-6">
        <b-button class="btn btn-primary" pill @click="$router.push('/admin-option-note-category-edit/new')">
          <i class="fas fa-plus-circle" aria-hidden="true" />
          Create New
        </b-button>
      </div>
      <div class="col-md-6">
        <div id="DataTables_Table_0_filter" class="dataTables_filter float-right">
          <label>Search:<input type="search" class="form-control form-control-sm" placeholder=""></label>
        </div>
      </div>
    </div>
    <table class="noSearch noSort clickable zebra table mt-4 schedule dataTable no-footer">
      <thead>
        <tr role="row">
          <th
            class="dt-center sorting_asc"
            tabindex="0"
            rowspan="1"
            colspan="1"
            aria-sort="ascending"
            aria-label="Category: activate to sort column descending"
            style="width: 262px;"
          >
            Category
          </th><th
            class="dt-center sorting"
            tabindex="0"
            rowspan="1"
            colspan="1"
            aria-label="Sub Category: activate to sort column ascending"
            style="width: 259.6px;"
          >
            Sub Category
          </th><th
            class="dt-center sorting"
            tabindex="0"
            rowspan="1"
            colspan="1"
            aria-label="Descriptor: activate to sort column ascending"
            style="width: 262px;"
          >
            Descriptor
          </th><th
            class="dt-center sorting"
            tabindex="0"
            rowspan="1"
            colspan="1"
            aria-label="Action: activate to sort column ascending"
            style="width: 210px;"
          >
            Action
          </th><th
            class="dt-center sorting"
            tabindex="0"
            rowspan="1"
            colspan="1"
            aria-label="Active: activate to sort column ascending"
            style="width: 62px;"
          >
            Active
          </th>
        </tr>
      </thead>
      <tbody>
        <tr role="row">
          <td class="dt-center sorting_1">
            <nuxt-link to="/admin-option-note-category-edit">
              Academics
            </nuxt-link>
          </td>
          <td class=" dt-center">
            <nuxt-link to="/admin-option-note-category-edit">
              Coursework
            </nuxt-link>
          </td>
          <td class=" dt-center">
            <nuxt-link to="/admin-option-note-category-edit">
              Progress
            </nuxt-link>
          </td>
          <td class=" dt-center">
            <nuxt-link to="/admin-option-note-category-edit">
              Goal setting
            </nuxt-link>
          </td>
          <td class=" dt-center">
            <span class="hide" style="display: none;">1</span><i class="icon active fas fa-circle" />
          </td>
        </tr>
        <tr role="row">
          <td class="dt-center sorting_1">
            <nuxt-link to="/admin-option-note-category-edit">
              Academics
            </nuxt-link>
          </td>
          <td class=" dt-center">
            <nuxt-link to="/admin-option-note-category-edit">
              Coursework
            </nuxt-link>
          </td>
          <td class=" dt-center">
            <nuxt-link to="/admin-option-note-category-edit">
              Progress
            </nuxt-link>
          </td>
          <td class=" dt-center">
            <nuxt-link to="/admin-option-note-category-edit">
              Goal setting
            </nuxt-link>
          </td>
          <td class=" dt-center">
            <span class="hide" style="display: none;">1</span><i class="icon active fas fa-circle" />
          </td>
        </tr>
        <tr role="row">
          <td class="dt-center sorting_1">
            <nuxt-link to="/admin-option-note-category-edit">
              Academics
            </nuxt-link>
          </td>
          <td class=" dt-center">
            <nuxt-link to="/admin-option-note-category-edit">
              Coursework
            </nuxt-link>
          </td>
          <td class=" dt-center">
            <nuxt-link to="/admin-option-note-category-edit">
              Progress
            </nuxt-link>
          </td>
          <td class=" dt-center">
            <nuxt-link to="/admin-option-note-category-edit">
              Goal setting
            </nuxt-link>
          </td>
          <td class=" dt-center">
            <span class="hide" style="display: none;">1</span><i class="icon active fas fa-circle" />
          </td>
        </tr>
        <tr role="row">
          <td class="dt-center sorting_1">
            <nuxt-link to="/admin-option-note-category-edit">
              Academics
            </nuxt-link>
          </td>
          <td class=" dt-center">
            <nuxt-link to="/admin-option-note-category-edit">
              Coursework
            </nuxt-link>
          </td>
          <td class=" dt-center">
            <nuxt-link to="/admin-option-note-category-edit">
              Progress
            </nuxt-link>
          </td>
          <td class=" dt-center">
            <nuxt-link to="/admin-option-note-category-edit">
              Goal setting
            </nuxt-link>
          </td>
          <td class=" dt-center">
            <span class="hide" style="display: none;">1</span><i class="icon active fas fa-circle" />
          </td>
        </tr>
        <tr role="row">
          <td class="dt-center sorting_1">
            <nuxt-link to="/admin-option-note-category-edit">
              Academics
            </nuxt-link>
          </td>
          <td class=" dt-center">
            <nuxt-link to="/admin-option-note-category-edit">
              Coursework
            </nuxt-link>
          </td>
          <td class=" dt-center">
            <nuxt-link to="/admin-option-note-category-edit">
              Progress
            </nuxt-link>
          </td>
          <td class=" dt-center">
            <nuxt-link to="/admin-option-note-category-edit">
              Goal setting
            </nuxt-link>
          </td>
          <td class=" dt-center">
            <span class="hide" style="display: none;">1</span><i class="icon active fas fa-circle" />
          </td>
        </tr>
        <tr role="row">
          <td class="dt-center sorting_1">
            <nuxt-link to="/">
              Academics
            </nuxt-link>
          </td>
          <td class=" dt-center">
            <nuxt-link to="/">
              Coursework
            </nuxt-link>
          </td>
          <td class=" dt-center">
            <nuxt-link to="/">
              Progress
            </nuxt-link>
          </td>
          <td class=" dt-center">
            <nuxt-link to="/">
              Goal setting
            </nuxt-link>
          </td>
          <td class=" dt-center">
            <span class="hide" style="display: none;">1</span><i class="icon active fas fa-circle" />
          </td>
        </tr>
        <tr role="row">
          <td class="dt-center sorting_1">
            <nuxt-link to="/">
              Academics
            </nuxt-link>
          </td>
          <td class=" dt-center">
            <nuxt-link to="/">
              Coursework
            </nuxt-link>
          </td>
          <td class=" dt-center">
            <nuxt-link to="/">
              Progress
            </nuxt-link>
          </td>
          <td class=" dt-center">
            <nuxt-link to="/">
              Goal setting
            </nuxt-link>
          </td>
          <td class=" dt-center">
            <span class="hide" style="display: none;">1</span><i class="icon active fas fa-circle" />
          </td>
        </tr>
        <tr role="row">
          <td class="dt-center sorting_1">
            <nuxt-link to="/">
              Academics
            </nuxt-link>
          </td>
          <td class=" dt-center">
            <nuxt-link to="/">
              Coursework
            </nuxt-link>
          </td>
          <td class=" dt-center">
            <nuxt-link to="/">
              Progress
            </nuxt-link>
          </td>
          <td class=" dt-center">
            <nuxt-link to="/">
              Goal setting
            </nuxt-link>
          </td>
          <td class=" dt-center">
            <span class="hide" style="display: none;">1</span><i class="icon active fas fa-circle" />
          </td>
        </tr>
        <tr role="row">
          <td class="dt-center sorting_1">
            <nuxt-link to="/">
              Academics
            </nuxt-link>
          </td>
          <td class=" dt-center">
            <nuxt-link to="/">
              Coursework
            </nuxt-link>
          </td>
          <td class=" dt-center">
            <nuxt-link to="/">
              Progress
            </nuxt-link>
          </td>
          <td class=" dt-center">
            <nuxt-link to="/">
              Goal setting
            </nuxt-link>
          </td>
          <td class=" dt-center">
            <span class="hide" style="display: none;">1</span><i class="icon active fas fa-circle" />
          </td>
        </tr>
        <tr role="row" class="odd">
          <td class="dt-center">
            <nuxt-link to="/">
              Accommodations
            </nuxt-link>
          </td><td class=" dt-center">
            <nuxt-link to="/" />
          </td>
          <td class=" dt-center">
            <nuxt-link to="" />
          </td>
          <td class=" dt-center">
            <nuxt-link to="/" />
          </td>
          <td class="dt-center sorting_1">
            <span class="hide" style="display: none;">0</span><i class="icon inactive fas fa-ban" />
          </td>
        </tr>
        <tr role="row" class="odd">
          <td class="dt-center">
            <nuxt-link to="/">
              Accommodations
            </nuxt-link>
          </td><td class=" dt-center">
            <nuxt-link to="/" />
          </td>
          <td class=" dt-center">
            <nuxt-link to="" />
          </td>
          <td class=" dt-center">
            <nuxt-link to="/" />
          </td>
          <td class="dt-center sorting_1">
            <span class="hide" style="display: none;">0</span><i class="icon inactive fas fa-ban" />
          </td>
        </tr>
        <tr role="row" class="odd">
          <td class="dt-center">
            <nuxt-link to="/">
              Accommodations
            </nuxt-link>
          </td><td class=" dt-center">
            <nuxt-link to="/" />
          </td>
          <td class=" dt-center">
            <nuxt-link to="" />
          </td>
          <td class=" dt-center">
            <nuxt-link to="/" />
          </td>
          <td class="dt-center sorting_1">
            <span class="hide" style="display: none;">0</span><i class="icon inactive fas fa-ban" />
          </td>
        </tr>
        <tr role="row" class="odd">
          <td class="dt-center">
            <nuxt-link to="/">
              Accommodations
            </nuxt-link>
          </td><td class=" dt-center">
            <nuxt-link to="/" />
          </td>
          <td class=" dt-center">
            <nuxt-link to="" />
          </td>
          <td class=" dt-center">
            <nuxt-link to="/" />
          </td>
          <td class="dt-center sorting_1">
            <span class="hide" style="display: none;">0</span><i class="icon inactive fas fa-ban" />
          </td>
        </tr>
        <tr role="row" class="odd">
          <td class="dt-center">
            <nuxt-link to="/">
              Accommodations
            </nuxt-link>
          </td><td class=" dt-center">
            <nuxt-link to="/" />
          </td>
          <td class=" dt-center">
            <nuxt-link to="" />
          </td>
          <td class=" dt-center">
            <nuxt-link to="/" />
          </td>
          <td class="dt-center sorting_1">
            <span class="hide" style="display: none;">0</span><i class="icon inactive fas fa-ban" />
          </td>
        </tr>
        <tr role="row" class="odd">
          <td class="dt-center">
            <nuxt-link to="/">
              Accommodations
            </nuxt-link>
          </td><td class=" dt-center">
            <nuxt-link to="/" />
          </td>
          <td class=" dt-center">
            <nuxt-link to="" />
          </td>
          <td class=" dt-center">
            <nuxt-link to="/" />
          </td>
          <td class="dt-center sorting_1">
            <span class="hide" style="display: none;">0</span><i class="icon inactive fas fa-ban" />
          </td>
        </tr>
        <tr role="row" class="odd">
          <td class="dt-center">
            <nuxt-link to="/">
              Accommodations
            </nuxt-link>
          </td><td class=" dt-center">
            <nuxt-link to="/" />
          </td>
          <td class=" dt-center">
            <nuxt-link to="" />
          </td>
          <td class=" dt-center">
            <nuxt-link to="/" />
          </td>
          <td class="dt-center sorting_1">
            <span class="hide" style="display: none;">0</span><i class="icon inactive fas fa-ban" />
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</template>

<script>
export default {
  name: 'Table'
}
</script>

<style>
.fa-circle{
    color: #2AE809;
}
</style>
