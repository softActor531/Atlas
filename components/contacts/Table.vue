<template>
  <div id="accordion" class="card mt-3">
    <div id="headingOne" class="card-header">
      <h4 class="mt-2 float-left">
        <a
          class="text-dark bold"
          data-toggle="collapse"
          href="#collapseOne"
          aria-expanded="true"
        >
          <i class="mdi mdi-minus mr-1 text-primary" />
          Contacts
        </a>
      </h4>
      <div class="m-0 float-right">
        <a
          href="#"
          class="btn btn-blue waves-effect waves-light"
          target="_blank"
        >Export</a>
        <a
          class="btn btn-blue waves-effect waves-light"
          href="#"
        >Add New Contact</a>
        <button
          class="btn-multiple-delete btn btn-info waves-effect waves-light"
        >
          Delete
        </button>
      </div>
    </div>
    <div
      id="collapseOne"
      class="collapse show"
      aria-labelledby="headingOne"
      data-parent="#accordion"
      style
    >
      <div class="card-body">
        <div
          id="contacts-table"
          class="table-responsive"
          data-pattern="priority-columns"
        >
          <div class="dataTables_wrapper">
            <div id="DataTables_Table_10_length" class="dataTables_length">
              <label>
                Show entries
                <select
                  class="custom-select custom-select-sm form-control form-control-sm contacts-length-menu"
                >
                  <option value="10">10</option>
                  <option selected="" value="25">25</option>
                  <option value="50">50</option>
                  <option value="100">100</option>
                </select>
              </label>
            </div>
          </div>
          <table class="table table-sm contacts-table text-center">
            <thead>
              <tr>
                <th>First Name</th>
                <th>Last Name</th>
                <th>Phase</th>
                <th>Status</th>
                <th>Action Count</th>
                <th>District</th>
                <th>District ID</th>
                <th>Academy ID</th>
                <th>Created Date</th>
                <th>Outreach Resource Advocate</th>
                <th>Community Outreach Advocate</th>
                <th>Priority</th>
                <th>Control</th>

                <th>
                  <button id class="btn-select-all">
                    Select All
                  </button>
                  <input id="select-all-btn" type="hidden" value="0">
                </th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <td class="text-center">
                  <a href="#" class="text-secondary" target="_blank">Brandon</a>
                </td>
                <td class="text-center">
                  <a href="#" class="text-secondary" target="_blank">Waugh</a>
                </td>
                <td class="text-center">
                  Recruitment
                </td>
                <td class="text-center">
                  Scheduled
                </td>
                <td class="text-center">
                  0
                </td>
                <td class="text-center">
                  Sarasota
                </td>
                <td class="text-center" />
                <td class="text-center">
                  SAR
                </td>
                <td class="text-center">
                  07/24/2020
                </td>
                <td class="text-center">
                  Jesse Corona
                </td>
                <td class="text-center" />
                <td class="text-center">
                  Normal
                </td>
                <td class="text-center actions">
                  <div class="btn-group btn-group-sm">
                    <a
                      href="#"
                      class="tabledit-edit-button btn-show text-secondary"
                    >
                      <i class="fas fa-eye" />
                    </a>
                  </div>
                  <div class="btn-group btn-group-sm">
                    <a
                      href="#"
                      class="tabledit-edit-button btn-edit text-secondary"
                    >
                      <i class="mdi mdi-pencil" />
                    </a>
                  </div>
                </td>

                <td class="text-center">
                  <input type="checkbox" name="selected" value>
                </td>
              </tr>
              <tr>
                <td class="text-center">
                  <a href="#" class="text-secondary" target="_blank">Brandon</a>
                </td>
                <td class="text-center">
                  <a href="#" class="text-secondary" target="_blank">Waugh</a>
                </td>
                <td class="text-center">
                  Recruitment
                </td>
                <td class="text-center">
                  Scheduled
                </td>
                <td class="text-center">
                  0
                </td>
                <td class="text-center">
                  Sarasota
                </td>
                <td class="text-center" />
                <td class="text-center">
                  SAR
                </td>
                <td class="text-center">
                  07/24/2020
                </td>
                <td class="text-center">
                  Jesse Corona
                </td>
                <td class="text-center" />
                <td class="text-center">
                  Normal
                </td>
                <td class="text-center">
                  <div class="btn-group btn-group-sm">
                    <a
                      href="#"
                      class="tabledit-edit-button btn-show text-secondary"
                    >
                      <i class="fas fa-eye" />
                    </a>
                  </div>
                  <div class="btn-group btn-group-sm">
                    <a
                      href="#"
                      class="tabledit-edit-button btn-edit text-secondary"
                    >
                      <i class="mdi mdi-pencil" />
                    </a>
                  </div>
                </td>

                <td class="text-center actions">
                  <input type="checkbox" name="selected" value>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </div>
</template>

<script>
export default {
  name: 'Table'
}
</script>

<style scoped></style>
