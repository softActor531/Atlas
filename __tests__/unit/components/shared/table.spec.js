import { shallowMount, createLocalVue } from '@vue/test-utils'
import Table from '@/components/shared/Table.vue'
import { BootstrapVue } from 'bootstrap-vue'

let wrapper

describe('Table', () => {
  beforeEach(() => {
    const localVue = createLocalVue()
    localVue.use(BootstrapVue)

    wrapper = shallowMount(Table, {
      localVue,
      data: () => ({}),
      props: {
        items: {
          type: Array,
          default: () => []
        }
      }
    })
  })

  test('is a Vue instance', () => {
    expect(wrapper.vm).toBeTruthy()
  })

  test('should load table', async () => {
    const table = wrapper.find('b-table-stub')
    expect(wrapper.html()).toMatchSnapshot()
    expect(table.exists()).toBe(true)
  })

  test('should load items', async () => {
    const table = wrapper.find('b-table-stub')

    wrapper.setProps({
      enableSearch: true,
      items: [
        {
          type: 'users',
          id: '1',
          first_name: 'Ahmed',
          last_name: 'Omran',
          title: 'Mr.',
          email: 'ahmed.omran@sixtysixten.com',
          'created-at':
        '2020-09-03T00:16:37.000000Z',
          links: {
            self: 'http://atlas-rebuild-api.mtcdevsite.com/api/v1/users/1'
          },
          role: 'Sales'
        },
        {
          type: 'users',
          id: '2',
          first_name: 'Karim',
          last_name: 'Hisham',
          title: 'Mr.',
          email: 'karim.hisham@sixtysixten.com',
          'created-at':
          '2020-09-03T00:26:42.000000Z',
          links: {
            self: 'http://atlas-rebuild-api.mtcdevsite.com/api/v1/users/2'
          },
          role: 'Scanner'
        }
      ]
    })
    await wrapper.vm.$nextTick()

    expect(table.props().items.length).toEqual(2)
  })

  test('should test computed', async () => {
    wrapper.setProps({
      fields: [{ key: 'first_name', label: 'First', sortable: true }]
    })
    await wrapper.vm.$nextTick()
    expect(wrapper.vm.sortOptions).toStrictEqual([{ text: 'First', value: 'first_name' }])
  })
})
