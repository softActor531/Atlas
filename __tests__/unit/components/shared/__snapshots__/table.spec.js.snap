// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table should load table 1`] = `
"<b-card-stub bodytag=\\"div\\" titletag=\\"h4\\" subtitletag=\\"h6\\" subtitletextvariant=\\"muted\\" headertag=\\"div\\" footertag=\\"div\\" tag=\\"div\\">
  <!---->
  <b-container-stub tag=\\"div\\">
    <b-row-stub tag=\\"div\\">
      <!---->
      <b-col-stub sm=\\"5\\" md=\\"4\\" tag=\\"div\\" class=\\"my-1\\">
        <b-form-group-stub label=\\"show\\" labelfor=\\"perPageSelect\\" labelsize=\\"md\\" labelcolssm=\\"8\\" labelcolsmd=\\"4\\" labelcolslg=\\"3\\" labelalignmd=\\"right\\" feedbackarialive=\\"assertive\\" class=\\"mb-0\\">
          <b-form-select-stub id=\\"perPageSelect\\" size=\\"sm\\" options=\\"25,50,100\\" valuefield=\\"value\\" textfield=\\"text\\" htmlfield=\\"html\\" disabledfield=\\"disabled\\" labelfield=\\"label\\" optionsfield=\\"options\\" value=\\"25\\" selectsize=\\"0\\"></b-form-select-stub>
          entries
        </b-form-group-stub>
      </b-col-stub>
    </b-row-stub>
    <b-row-stub tag=\\"div\\">
      <b-table-stub items=\\"\\" value=\\"\\" small=\\"true\\" responsive=\\"true\\" stacked=\\"md\\" filterincludedfields=\\"\\" filterdebounce=\\"0\\" sortby=\\"\\" sortdirection=\\"asc\\" sortcompareoptions=\\"[object Object]\\" labelsortasc=\\"Click to sort Ascending\\" labelsortdesc=\\"Click to sort Descending\\" labelsortclear=\\"Click to clear sorting\\" sorticonleft=\\"true\\" perpage=\\"25\\" currentpage=\\"1\\" selectmode=\\"multi\\" selectedvariant=\\"active\\" showempty=\\"true\\" emptytext=\\"There are no records to show\\" emptyfilteredtext=\\"There are no records matching your request\\" apiurl=\\"\\" custom-prop=\\"any\\"></b-table-stub>
    </b-row-stub>
    <b-row-stub tag=\\"div\\">
      <b-col-stub lg=\\"6\\" tag=\\"div\\" class=\\"my-1\\">
        Showing 0 to
        0 of 0 entries (filtered from
        0 total entries)
      </b-col-stub>
      <b-col-stub lg=\\"6\\" tag=\\"div\\" class=\\"my-1\\">
        <b-pagination-stub value=\\"1\\" limit=\\"5\\" align=\\"right\\" pills=\\"true\\" arialabel=\\"Pagination\\" labelfirstpage=\\"Go to first page\\" firsttext=\\"«\\" labelprevpage=\\"Go to previous page\\" prevtext=\\"‹\\" labelnextpage=\\"Go to next page\\" nexttext=\\"›\\" labellastpage=\\"Go to last page\\" lasttext=\\"»\\" labelpage=\\"Go to page\\" ellipsistext=\\"…\\" perpage=\\"25\\" totalrows=\\"0\\" class=\\"my-3 heavy-rain-gradient\\"></b-pagination-stub>
      </b-col-stub>
    </b-row-stub>
  </b-container-stub>
</b-card-stub>"
`;
